ksqldb-udf/src/main/java/io/confluent/ksql/types/KsqlStruct.java

```diff
@@ -35,36 +39,57 @@
 @Immutable
 public final class KsqlStruct {
 
-  private final SqlStruct schema;
   @EffectivelyImmutable
   private final ImmutableList<Optional<?>> values;
 
   public static Builder builder(final SqlStruct schema) {
     return new Builder(schema);
   }
 
-  private KsqlStruct(
-      final SqlStruct schema,
-      final List<Optional<?>> values
-  ) {
-    this.schema = Objects.requireNonNull(schema, "schema");
+  private KsqlStruct(final List<Optional<?>> values) {
```

Comment:
I feel like if we're touting efficiency in this class, we should strive not to use optionals to avoid unnecessary wrapping/unwrapping - but that can be change for another time :)

Reply:
I'd be surprised if it made a significant difference.  We'd need to profile to work that out.